src/Core/QueryT.cs

```diff
@@ -94,6 +94,9 @@ select GetResult(context).Select(e => e.Value)
 
         public Query<T> Take(int count) =>
             Bind(xs => Query.Return(xs.Take(count)));
+        
+        public Query<T> Distinct =>
```

Comment:
Turn this into a method and add an overload that enables one to supply an equality comparer, just as you have with [`Enumerable.Distinct`](https://msdn.microsoft.com/en-us/library/bb338049(v=vs.110).aspx).